stdlib/asm/math/u64.masm

```diff
@@ -442,8 +491,55 @@ end
 # Stack transition looks as follows:
 # [b_hi, b_lo, a_hi, a_lo, ...] -> [c_hi, c_lo, ...], where c = a // b
 export.checked_div
-    exec.u32assert4
-    exec.unchecked_div
+    adv.u64div          # inject the quotient and the remainder into the advice tape
+
+    push.adv.1
+    push.adv.1
+    u32assert.2
+
+    dup.3               # multiply quotient by the divisor and make sure the resulting value
+    dup.2               # fits into 2 32-bit limbs
+    u32mul.unsafe
+    dup.4
+    dup.4
+    u32madd.unsafe
+    eq.0
+    assert
+    dup.5
+    dup.3
+    u32madd.unsafe
+    eq.0
+    assert
+    dup.4
+    dup.3
+    mul
+    eq.0
+    assert
```

Comment:
In this part, we should probably check that both quotient and the divisor are valid u64 values (right now, it seems like we are checking only the quotient).